### PR TITLE
Variants to table NPE

### PR DESCRIPTION
**Pull request by bbimber**
### Description

I encountered an NPE when trying to run GATK's VariantsToTable when requesting the genotype filter field ("-GF FT").  At VariantsToTable line 334, we call Genotype.getAnyAttribute().  At Genotype line 346, you'll see when requesting "FT", it returns the values of Genotype.getFilters(), which can be NULL if the genotype is unfiltered.  If a NULL value is then passed to VariantsToTable.prettyPrintObject(), it will NPE.  This pull request adds a simple null check and returns empty string if null.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below. 

#### Content
- [ ] Added or modified tests to cover changes and any new functionality
- [ ] All tests passing on Bamboo (including license tests!)

#### Form
- [ ] Rebased, squashed and reworded to produce a single commit (exemptions may be made -- sparingly) that is _uncluttered by excess lines_ (lines that just say "addressed review", "fixed" etc left over from squashing)
- [ ] Both the PR and the final rebased commit have a **concise** yet **descriptive** title (they are used when we compile version notes)
- [ ] Edited the README / documentation accordingly and got sign-off from support team (private code is exempt)

#### Review
- [ ] Suggest a reviewer or ask your team lead to suggest one
- [ ] Final (thumbsup) from the reviewer(s)

#### GATK4 Implications
- [ ] If your fix/change is applicable to GATK4 as well, and is reasonably small and self-contained (< ~50 lines or so), port the change to GATK4 and open a PR against https://github.com/broadinstitute/gatk or https://github.com/broadinstitute/gatk-protected as appropriate, or at least make a "best effort" attempt to do so.
- [ ] If your fix/change cannot yet be ported to GATK4 because the tool in question hasn't been ported yet, or has only been partially ported, or it would be difficult/burdensome to port the change, or you tried to port the change and failed, then add the ticket to our list of [GATK3 PRs to be eventually ported to GATK4](https://docs.google.com/document/d/1DjEHw57k5h0i8MZRGYPlQA3InvURKwQ7pCoi_Eigc4M/edit)

Once everything is checked off, you can go ahead and merge the PR. Don't forget to also delete the branch.

### CODE REVIEW

**Comment by vdauwera** [comment]
Hey @bbimber, sorry for the lag, we just noticed this. This seems straightforward, thank you. 

@ronlevine Can you please port this to gsa-unstable? (without the pom file)

File: public/discvr-seq/src/main/java/org/labkey/discvrseq/Walker1.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    public Object reduceInit() {
+        return null;
+    }
+
+    @Override
+    public Object reduce(Object o, Object o2) {
+        return null;
+    }
+}
+
```
Comment (paulmenzel) [inline public/discvr-seq/src/main/java/org/labkey/discvrseq/Walker1.java]: Remove the blank line at the end?

**Review by paulmenzel — COMMENTED** [review]
Why is the merge commit needed?

**Comment by ronlevine** [comment]
pom.xml and Walker1.java appear to be extraneous files. This pull request needs tests. I believe a full fix is a port of https://github.com/broadinstitute/gsa-unstable/pull/1606.

**Comment by paulmenzel** [comment]
Unfortunately, that repository is private (not public).

**Comment by ronlevine** [comment]
@paulmenzel Sorry, I forgot it's private. I will port the fix and notify you once it's done.

**Comment by bbimber** [comment]
yes, sorry about the unclean PR.  this is such an old PR and I may have added some junk after without thinking afterwards.  i can remake if you want.  this should be a 3 line change to VariantsToTable only.

**Comment by ronlevine** [comment]
I think it should print "NA" instead of "" for this use case. It also needs tests.

**Comment by ronlevine** [comment]
Please add tests. I think the missing columns should be `NA` instead of blank.

**Comment by ronlevine** [comment]
@bbimber @paulmenzel Take a look at https://github.com/broadinstitute/gatk/pull/3252. This fixes the NPE and puts NA in the columns if an item is missing.

**Comment by bbimber** [comment]
Yes, that seems like a reasonable route and it would be OK to close out this PR.

I did end up writing a very simple unit test this morning.  I'm just attaching a patch here and will leave it up to the GATK devs if you want to add it to that issue.  Note: you'd probably want to marginally tweak the uses to match the behavior of #3252, but it will be really close.

[VariantsToTableUnitTest.patch.txt](https://github.com/broadgsa/gatk/files/1139820/VariantsToTableUnitTest.patch.txt)

**Comment by ronlevine** [comment]
Refer to https://github.com/broadinstitute/gatk/pull/3190 for the fix. I closed https://github.com/broadinstitute/gatk/pull/3252 since it was a duplicate.

**Comment by ronlevine** [comment]
@bbimber Can this be closed?

**Comment by bbimber** [comment]
ok

**Pull request closed without merging**